x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities.tsx

```diff
@@ -321,112 +303,146 @@ const VulnerabilitiesContent = ({ dataView }: { dataView: DataView }) => {
+                      values: { total: data?.total },
+                    })}
+                  </EuiButtonEmpty>
+                </>
+              ),
+            },
+            right: (
+              <EuiFlexItem grow={false} className={styles.groupBySelector}>
+                <FindingsGroupBySelector
+                  type="default"
+                  pathnameHandler={vulnerabilitiesPathnameHandler}
+                />
+              </EuiFlexItem>
+            ),
+          },
+        }}
+        gridStyle={{
+          border: 'horizontal',
+          cellPadding: 'l',
+          stripes: false,
+          rowHover: 'none',
+          header: 'underline',
+        }}
+        renderCellValue={renderCellValue}
+        inMemory={{ level: 'enhancements' }}
+        sorting={{ columns: sort, onSort: onSortHandler }}
+        pagination={{
+          pageIndex,
+          pageSize,
+          pageSizeOptions: [10, 25, 100],
+          onChangeItemsPerPage,
+          onChangePage,
+        }}
+      />
+      {isLastLimitedPage && <LimitedResultsBar />}
+      {showVulnerabilityFlyout && selectedVulnerability && (
+        <VulnerabilityFindingFlyout
+          flyoutIndex={selectedVulnerabilityIndex}
+          vulnerabilityRecord={selectedVulnerability}
+          totalVulnerabilitiesCount={limitedTotalItemCount}
+          onPaginate={onPaginateFlyout}
+          closeFlyout={onCloseFlyout}
+          isLoading={isFetching}
+        />
+      )}
+    </>
+  );
+};
+const VulnerabilitiesContent = ({ dataView }: { dataView: DataView }) => {
+  const { pageIndex, query, sort, queryError, pageSize, setUrlQuery } = useCloudPostureTable({
+    dataView,
+    defaultQuery: getDefaultQuery,
+    paginationLocalStorageKey: LOCAL_STORAGE_PAGE_SIZE_FINDINGS_KEY,
+  });
+
+  const multiFieldsSort = useMemo(() => {
+    return sort.map(({ id, direction }: { id: string; direction: string }) => {
+      if (id === vulnerabilitiesColumns.severity) {
+        return severitySortScript(direction);
+      }
+      if (id === vulnerabilitiesColumns.package) {
+        return getCaseInsensitiveSortScript(id, direction);
+      }
+
+      return {
+        [id]: direction,
+      };
+    });
+  }, [sort]);
+
+  const { data, isLoading, isFetching } = useLatestVulnerabilities({
+    query,
+    sort: multiFieldsSort,
+    enabled: !queryError,
+    pageIndex,
+    pageSize,
+  });
+
+  const error = queryError || null;
+
+  if (isLoading && !error) {
     return defaultLoadingRenderer();
   }
 
-  if (!data?.page) {
+  if (!data?.page && !error) {
     return defaultNoDataRenderer();
   }
 
-  const showVulnerabilityFlyout = flyoutVulnerabilityIndex > invalidIndex;
-
   return (
     <>
       <FindingsSearchBar
         dataView={dataView}
         setQuery={(newQuery) => {
           setUrlQuery({ ...newQuery, pageIndex: 0 });
         }}
-        loading={isLoading}
+        loading={isFetching}
```

Comment:
fix for the search bar loading state. The component exits early on `isLoading`, so `isFetching` should be used the same way as in Misconfigurations